src/components/Discussion/Internal/Comment/CommentActions.tsx

```diff
@@ -26,30 +29,54 @@ const styles = {
   })
 }
 
-export const Actions = ({ t, comment, onExpand, onReply }) => {
-  const { published } = comment
-  const { discussion, actions, clock } = React.useContext(DiscussionContext)
-  const { displayAuthor, userWaitUntil, userCanComment } = discussion
+const propTypes = {
+  t: PropTypes.func.isRequired,
+  comment: PropTypes.object.isRequired,
+  actions: PropTypes.shape({
+    handleLoadReplies: PropTypes.func,
+    handleReply: PropTypes.func,
+    handleShare: PropTypes.func
+  }),
+  voteActions: PropTypes.shape({
+    handleUpVote: PropTypes.func.isRequired,
+    handleDownVote: PropTypes.func.isRequired,
+    handleUnVote: PropTypes.func.isRequired
+  }),
+  userCanComment: PropTypes.bool,
+  userWaitUntil: PropTypes.string,
+  isBoard: PropTypes.bool
+}
+
+export const CommentActions = ({
+  t,
+  comment,
+  actions: { handleLoadReplies, handleReply, handleShare },
+  voteActions,
+  userCanComment,
+  userWaitUntil,
+  isBoard
+}) => {
+  const isDesktop = useMediaQuery(mUp)
 
   const now = useCurrentMinute()
   const [colorScheme] = useColorContext()
 
-  const replyBlockedMessage = useMemo(() => {
+  const replyBlockedMessage: string | null = useMemo(() => {
     const waitUntilDate = userWaitUntil && new Date(userWaitUntil)
-    if (waitUntilDate && waitUntilDate > now) {
+    if (waitUntilDate && waitUntilDate.getTime() > now) {
```

Comment:
typescript lead to a wrong solution initial here:
https://github.com/orbiting/styleguide/pull/469/commits/5838406169858406e6baf22979e9cea05e372b9a

typecasting was fine:
<img width="1278" alt="Screenshot 2022-01-25 at 17 36 06" src="https://user-images.githubusercontent.com/410211/151019281-6f6556ae-cf94-4541-836f-04c3ea300125.png">

(of course this also has to do with bad naming in the JS dates api but still typescript sent you in a wrong direction)